src/lib/reasoners/enum_rel.ml

```diff
@@ -244,13 +272,22 @@ let assume env uf la =
 
 let add env _ r _ = add_aux env r, []
 
+(* Do a case-split by choosing a value for a uninterpreted semantic value
+   whose the domain in [env] is of minimal size. *)
 let case_split env uf ~for_model =
   let acc = MX.fold
       (fun r (hss, _) acc ->
          let x, _ = Uf.find_r uf r in
          match Sh.embed x with
-         | Cons _ -> acc (* already bound to an Enum const *)
-         | _ -> (* cs even if sz below is equal to 1 *)
+         | Cons _ ->
+           (* The equivalence class of [r] already contains a value and
+              we don't need to make another case-split for this semantic
+              value. *)
+           acc
+         | _ ->
+           (* We have to do a case-split, even if the domain of [r] is
+              of cardinal 1 as we have to put this value in the
+              equivalence class of [r]. *)
```

Comment:
I think that "let `Uf` know about the equality" would be more precise — in particular this matters for model generation.